mpas_analysis/analysis_task_template.py

```diff
@@ -0,0 +1,217 @@
+   and files are present), and will finally call run on each task that is
+   to be generated and is set up properly.
+
+Don't forget to remove this docstring. (It's not needed.)
+
+Authors
+-------
+Xylar Asay-Davis
+'''
+
+# import python modules here
+
+# import mpas_analysis module here (those with relative paths starting with
+# dots)
+from ..shared.analysis_task import AnalysisTask
+
+
+# Everywhere in this template, change MyTask to the name of your task, starting
+# with a capital letter and using mixed case but no underscores.  Leave
+# "AnalysisTask" as it is.  This indicates that MyTask has AnalysisTask as its
+# parent, "inhereting" several variables and methods (functions) from it that
+# can be helpful in initializing, checking whether to perform analysis and
+# performing the analysis.  See AnalysisTask in shared/analysis_task.py
+class MyTask(AnalysisTask):  # {{{
+    '''
+    <Briefly describe the analysis task here.  Just a 1-2 sentence description
+    of what the task does should be sufficient.>
+
+    Authors
+    -------
+    <List of authors>
+    '''
+
+    # This function is the "constructor", which is called when you want to
+    # create a new object from your class.  Typically, it will just store a few
+    # useful pieces of information like the name of the task, the component and
+    # maybe some tags describing the task.
+    # Keep self and config arguments.  Either modify myArg to be a useful
+    # additional argument (e.g. the name of a field to be analyzed) or remove
+    # it.
+    def __init__(self, config, myArg='myDefaultValue'):  # {{{
+        '''
+        Construct the analysis task.
+        <Add any additional description of what happens during construction>
+
+        Parameters
+        ----------
+        config :  instance of MpasAnalysisConfigParser
+            Contains configuration options
+
+        myArg : str, optional
+            <Describe the arg, or just remove remove it>
+
+        Authors
+        -------
+        <List of authors>
+
+        '''
+        # first, call the constructor from the base class (AnalysisTask).
+        # Modify MyTask, "myTask", "component" and ['tag1', 'tag2'] below:
+        # taskName is the same as the class name but with a lowercase letter
+        # componentName is one of 'ocean' or 'seaIce' (same as the name of
+        #     the folder where the task resides)
+        # tags are some useful names describing the task ('timeSeries',
+        #     'climatology', 'horizontalMap', 'index', 'transect', etc.) that
+        #     can be used in the 'generate' config option, e.g.
+        #     'all_climatology' of 'no_transect'.  Tasks that include computing
+        # time series, indices and/or climatologies should include
+        # 'timeSeries', 'index' and/or 'climatology' tags, as these are helpful
+        # used to compute and update the start and end dates of each of these
+        # analysis tasks as part of setting up the task.
```

Comment:
very nice explanation above, thanks.
Similar to the comment about `self`, would it be OK to mention briefly what `super` does?